pkg/detector/detector.go

```diff
@@ -466,6 +473,12 @@ func (d *ResourceDetector) ApplyClusterPolicy(object *unstructured.Unstructured,
 		bindingCopy := binding.DeepCopy()
 		err = retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
 			operationResult, err = controllerutil.CreateOrUpdate(context.TODO(), d.Client, bindingCopy, func() error {
+				// If this binding exists and its owner is not the input object, return error and let garbage collector
+				// delete this binding and try again later. See https://github.com/karmada-io/karmada/issues/2090.
+				if ownerRef := metav1.GetControllerOfNoCopy(bindingCopy); ownerRef != nil && ownerRef.UID != object.GetUID() {
```

Comment:
I think we should add this for L516 as well?

nit: maybe `bindingCopy.Spec.Resource.UID != object.GetUID()` is shorter?

Reply:
> nit: maybe bindingCopy.Spec.Resource.UID != object.GetUID() is shorter?

It is. But I think we should use `ownerRef` because it is designed for representing the dependency.

Reply:
> I think we should add this for L516 as well?

Please @Garrybest check.

Reply:
> I think we should add this for L516 as well?
Indeed, thanks for reminding.

> nit: maybe bindingCopy.Spec.Resource.UID != object.GetUID() is shorter?
Nice finding👍. Both are OK but maybe using ownerRef is more distinct literally.